lib/application/pages/home/collections/update/update_collection_page.dart

```diff
@@ -71,17 +70,24 @@ class _UpdateCollectionContents extends ConsumerWidget {
     }
 
     if (state is UpdateCollectionLoaded) {
+      late Widget body;
+
       switch (selectedSegment) {
         case _Segment.details:
-          return ProviderScope(
-            overrides: [
-              updateDetailsMetadata.overrideWithValue(state.collectionMetadata),
-            ],
-            child: UpdateCollectionDetails(),
-          );
+          body = UpdateCollectionDetails();
+          break;
         case _Segment.memos:
-          return UpdateCollectionMemos();
+          body = UpdateCollectionMemos();
+          break;
       }
+
+      return ProviderScope(
```

Comment:
Why can't we provide the overrides to each specific scope that needs it? Like `updateDetailsMetadata` to `UpdateCollectionDetails` and `updateMemosMetadata` to `UpdateCollectionMemos`?

Reply:
Unfortunatelly not. Updating providers dynamically is not possible. See:

<details>
<summary>Code</summary>

```dart
switch (selectedSegment) {
        case _Segment.details:
          return ProviderScope(
            overrides: [
              updateDetailsMetadata.overrideWithValue(state.collectionMetadata),
            ],
            child: UpdateCollectionDetails(),
          );

          break;
        case _Segment.memos:
          return ProviderScope(
            overrides: [
              updateMemosMetadata.overrideWithValue(state.memosMetadata),
            ],
            child: UpdateCollectionMemos(),
          );
          break;
      }
```

</details>

<details>
<summary>Error</summary>

<img width="368" alt="CleanShot 2021-11-25 at 14 03 31@2x" src="https://user-images.githubusercontent.com/11745745/143480628-a0a57b7c-b940-4bf1-9e7b-e667d47d62c4.png">

</details>

Reply:
Have you tried using different keys? The error seems to be that you're changing the `ProviderScope` overrides, which is not true, you're just sending a completely new child.

Reply:
But in the code above the overrides are changing, from this:

```dart
overrides: [
              updateDetailsMetadata.overrideWithValue(state.collectionMetadata),
            ]
```

to this

```dart
overrides: [
              updateMemosMetadata.overrideWithValue(state.memosMetadata),
            ],
```

Reply:
And that would be fine with a new `key` wouldn't it?

Reply:
And that would be fine with a new `key` wouldn't it?

Reply:
Using keys worked like a charm